docs/developer/design/volume_v1.0.rst

```diff
@@ -0,0 +1,294 @@
+Resource Definition** (CRD), built on top of the existing concept of
+**Persistent Volume** from Kubernetes. Those **Custom Resources** (CR) will be
+managed by a dedicated Kubernetes operator which will be responsible for the
+storage preparation (using Salt states) and lifetime management of the backing
+**Persistent Volume**.
+
+Volume management will be available from the Platform UI (through a dedicated
+tab under the node page). There, users will be able to create, monitor and
+delete MetalK8s volumes.
+
+
+Scope
+-----
+
+The scope of this first version of Volume Management will be minimalist but
+still functionally useful.
+
+
+Goal
+^^^^
+
+* support two kinds of **Volume**: **sparseLoopDevice** (backed by a sparse
+  file) and **rawBlockDevice** (using whole disk)
+* add support for volume creation (one by one) in the Platform UI
+* add support for volume deletion (one by one) in the Platform UI
+* add support for volume listing/monitoring (show status, size, …) in the
+  Platform UI
+* document how to create a volume
+* document how to create a storage class
+* automated tests on volume workflow (creation, deletion, …)
+
+
+Non-Goal
+^^^^^^^^
+
+* RAID support
+* LVM support
+* expose raw block device (unformated) as **Volume**
+* use an **Admission Controller** for semantic validation
+* auto-discovery of the disks
+* batch provisioning from the UI
+
+
+Proposal
+--------
+
+To implement this feature we need to:
+
+* define and deploy a new CRD describing a MetalK8s Volume
+* develop and deploy a new Kubernetes operator to manage the MetalK8s volumes
+* develop new Salt states to prepare and cleanup underlying storage on the
+  nodes
+* update the Platform UI to allow volume management
+
+
+User stories
+^^^^^^^^^^^^
+
+
+Volume creation
+~~~~~~~~~~~~~~~
+
+As a user I need to be able to create MetalK8s volume from the Platform UI.
+
+At creation time and can specify the type of volume I want, and then either its
+size (for **sparseLoopDevice**) or the backing device (for **rawBlockDevice**).
+
+I should be able monitor the progression of the volume creation from the UI and
+see when the volume is ready to use (or if an error occured).
+
+
+Volume monitoring
+~~~~~~~~~~~~~~~~~
+
+As a user I should be able to see all the volume existing on a specified node
+as well as their state.
+
+
+Volume deletion
+~~~~~~~~~~~~~~~
+
+As a user I need to be able to delete a MetalK8s volume from the Platform UI
+when I no longer need it.
+
+The UI should prevent me to delete used Volume.
+
+I should be able monitor the progression of the volume deletion from the UI.
+
+
+Components interactions
+^^^^^^^^^^^^^^^^^^^^^^^
+
+User will create Metalk8s volume through the Platform UI.
+
+The Platform UI will create and delete **Volume** CRs from the API server.
+
+The operator will watch event related to **Volume** CRs and
+**PersistentVolume** owned by **Volume** and react in order to update the state
```

Comment:
```suggestion
**PersistentVolumes** owned by **Volumes** and react in order to update the state
```

Reply:
Done (a bit differently because I don't want to add `s` to the object name)